recipes/mbedtls/all/conanfile.py

```diff
@@ -53,22 +57,19 @@ def configure(self):
             raise ConanInvalidConfiguration("{}-{} is not supported by this recipe".format(self.settings.compiler, self.settings.compiler.version))
 
     def requirements(self):
-        if self.options.with_zlib:
+        if self.options.get_safe("with_zlib"):
             self.requires("zlib/1.2.11")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = "{}-{}".format(self.name, self._version)
-        if tools.Version(self.version) >= "2.23.0": # went to mbedtls-mbedtls-X.X.X
-            extracted_dir = "{}-{}".format(self.name, extracted_dir)
-        os.rename(extracted_dir, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version],
+                    strip_root = True, destination=self._source_subfolder)
 
     def _configure_cmake(self):
         if not self._cmake:
             self._cmake = CMake(self)
             self._cmake.definitions["USE_SHARED_MBEDTLS_LIBRARY"] = self.options.shared
             self._cmake.definitions["USE_STATIC_MBEDTLS_LIBRARY"] = not self.options.shared
-            self._cmake.definitions["ENABLE_ZLIB_SUPPORT"] = self.options.with_zlib
+            self._cmake.definitions["ENABLE_ZLIB_SUPPORT"] = self.options.get_safe("with_zlib") or False
```

Comment:
```suggestion
            if tools.Version(self.version) < "3.0.0":
                self._cmake.definitions["ENABLE_ZLIB_SUPPORT"] = self.options.with_zlib
```

To avoid to inject an unused CMake variable